### PR TITLE
Ready to release 2.2.0

### PR DESCRIPTION
**Pull request by yidongnan**
* Use spotless to format *.gradle and *.md and *.yml files
* Modify example and remove not used config
* Add gRPC Common Javadoc

### CODE REVIEW

**Comment by GregorEeckels** [comment]
Regarding the  [X509CertificateAuthentication.java](https://github.com/yidongnan/grpc-spring-boot-starter/blob/master/grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/security/authentication/X509CertificateAuthentication.java) and [X509CertificateAuthenticationProvider.java](https://github.com/yidongnan/grpc-spring-boot-starter/blob/master/grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/security/authentication/X509CertificateAuthenticationProvider.java). Doesnt Spring Security have X509 support built in already? Can't that be reused?

File: grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/config/GrpcServerProperties.java [hunk 1]
```diff
@@ -35,9 +38,17 @@
...
+    @Setter(AccessLevel.NONE)
+    public static final String ANY_IP_ADDRESS = "*";
+
+    @Getter(AccessLevel.NONE)
+    @Setter(AccessLevel.NONE)
+    public static final String ANY_IPv4_ADDRESS = "0.0.0.0";
+
+    @Getter(AccessLevel.NONE)
+    @Setter(AccessLevel.NONE)
+    public static final String ANY_IPv6_ADDRESS = "::";
```
Comment (yidongnan) [inline grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/config/GrpcServerProperties.java]: Disable to generate metadata for these fields. 
image

File: grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/config/GrpcServerProperties.java [hunk 2]
```diff
@@ -35,9 +35,11 @@
...
 public class GrpcServerProperties {
 
-    public static String ANY_IP_ADDRESS = "*";
-    public static String ANY_IPv4_ADDRESS = "0.0.0.0";
-    public static String ANY_IPv6_ADDRESS = "::";
+    public static final String ANY_IP_ADDRESS = "*";
+
+    public static final String ANY_IPv4_ADDRESS = "0.0.0.0";
+
+    public static final String ANY_IPv6_ADDRESS = "::";
```
[the remark this answers is not in the document]
Comment (ST-DDT) [reply grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/config/GrpcServerProperties.java]: AFAICT there aren't any metadata for these fields anymore (at least in `spring-configuration-metadata.json`)

[the remark this answers is not in the document]
Comment (ST-DDT) [reply grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/config/GrpcServerProperties.java]: I don't see a reason to make this field non-final. I probably forgot to make it final in the first place. 
With your changes applied, I don't see any need for additional changes, or am I missing something?

**Pull request merged**